### DISCUSSION TITLE
IRobustJob / IParallelRobustJob for threading

### DISCUSSION THREAD

**Discussion by metalgearsloth** [space-wizards/space-station-14#21967] [Code announcements]
IRobustJob and IParallelJob (to split out into several IRobustJobs) have been added to the engine. These can be passed to ParallelManager to have work run on the threadpool instead of relying upon Task.Run / Parallel.For and allocates less. It also easily allows you to change batch sizes and store the job across ticks to avoid re-allocating it every time.

**Comment by Brandon-Huu** [comment]
Pog
